File: LeetCode_442/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> answer, cnt(100001, 0);

        for (int x : nums) {
            cnt[x] += 1;

            if (cnt[x] == 2) {
                answer.push_back(x);
            }
        }
        return answer;
    }
};

File: LeetCode_442/Main.py
from typing import List


class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        answer = []
        cnt = [0 for _ in range(len(nums) + 1)]

        for num in nums:
            cnt[num] += 1

            if cnt[num] == 2:
                answer.append(num)
        return answer